uxarray/grid.py

```diff
@@ -372,6 +373,79 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
+    # A one-time-used helper function that will only be used in the build_face_edges_connectivity map() function usage
+    def __replace_fill_val(self, row, index, rep_val):
+        row[index] = rep_val
+        return row
+
+    def build_face_edges_connectivity(self):
+        """A DataArray of indices indicating edges that are neighboring each
+        face.
+
+        Notes
+        -----
+        This function will add `Grid.ds.Mesh2_face_edges` to the `Grid` class, which is an integer
+        DataArray of size (nMesh2_face, MaxNumNodesPerFace)
+        """
+        mesh2_face_nodes = self.ds["Mesh2_face_nodes"].values
+        n, m = mesh2_face_nodes.shape
+
+        # First identify the _FillValue in the mesh2_face_nodes
+        # We will unify the _FillValue used in this function as -1:
+        if np.isnan(_FillValue_):
+            mesh2_face_nodes = np.nan_to_num(mesh2_face_nodes, nan=-1)
+        elif _FillValue_ is not -1:
+            mesh2_face_nodes[mesh2_face_nodes == _FillValue_] = -1
+
+        # Then do the padding for each face to close the polygon
+        closed = np.full((n, m + 1), -1, dtype=np.intp)
+        closed[:, :-1] = np.array(mesh2_face_nodes, dtype=np.intp)
+        # We only want the index of first occurrence of -1
+        first_fill_value_index = np.array([
+            np.array(list(*np.where(row == -1)))[0] for row in closed
+        ])  # np.array(list(zip(*np.where(closed == -1))))
```

Comment:
Can replace this with an elegant one-liner (see [here](https://stackoverflow.com/questions/16243955/numpy-first-occurrence-of-value-greater-than-existing-value) for explanation) 
```Python
first_fill_value_index = np.argmax(closed == -1, axis=1)